posthog/api/dashboard.py

```diff
@@ -187,7 +189,18 @@ def get_queryset(self) -> QuerySet:
     def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> response.Response:
         pk = kwargs["pk"]
         queryset = self.get_queryset()
-        dashboard = get_object_or_404(queryset, pk=pk)
+        try:
+            dashboard = get_object_or_404(queryset, pk=pk)
+        except Http404 as e:
+            user = cast(User, request.user)
+            alternative_team = Dashboard.objects.filter(deleted=False, team__in=user.teams, pk=pk)
+            if alternative_team.exists():
+                raise ObjectExistsInOtherProject(
+                    alternative_team_id=alternative_team.first().team_id,  # type: ignore
```

Comment:
This queries Postgres twice unnecessarily. The same could be achieved with just a `.first()`:
```suggestion
            alternative_team_dashboard = Dashboard.objects.filter(deleted=False, team__in=user.teams, pk=pk).first()
            if alternative_team_dashboard is not None:
                raise ObjectExistsInOtherProject(
                    alternative_team_id=alternative_team_dashboard.team_id,  # type: ignore
```